### ISSUE TITLE
[bug] Pydantic model definition errors for fragments with nesting

### ISSUE THREAD

**Issue by fekstr** [mirumee/ariadne-codegen#253]
## Description

Pydantic types are incorrectly defined when fragments contain nested nodes.

Example:

```graphql
# works
fragment IssueFields on Issue {
    id
}

query getIssue($id: ID!) {
    node(id: $id) {
        ...IssueFields
    }
}

# works
query getIssue($id: ID!) {
    node(id: $id) {
        ...on Issue {
            id
            author {
                login
            }
        }
    }
}

# does not work
fragment IssueFields on Issue {
    id
    author {
        login
    }
}

query getIssue($id: ID!) {
    node(id: $id) {
        ...IssueFields
    }
}
```

There are two issues in the last case.

1. The generated `IssueFieldsAuthor` type includes a required `__typename` field that is not included in the generated query, causing a validation error.
2. The generated `IssueFields` type contains an `author` field of type `Optional["IssueFieldsAuthor"]` that somehow seems to be undefined at instantiation, yielding the following error: 
```
pydantic.errors.PydanticUserError: `GetIssue` is not fully defined; you should define `GetIssueNodeNode`, then call `GetIssue.model_rebuild()`
```

The first issue disappears when including `__typename` in the generated query (in the client `get_issue` method).

The second issue disappears when defining `IssueFieldsAuthor` before `IssueFields` in `fragments.py`.

## Reproduction

1. Define these queries

```graphql
# queries.graphql

fragment IssueFields on Issue {
    id
    author {
        login
    }
}

query getIssue($id: ID!) {
    node(id: $id) {
        ...IssueFields
    }
}
```

Use the [GitHub schema](https://docs.github.com/en/graphql/overview/public-schema).

Generate the types with `ariadne-codegen`.

Run the following:

```python
import asyncio

# use the path to the generated client here
from src.graphql.graphql_client import Client


async def run():
    access_token = "<YOUR TOKEN>"
    client = Client(
        url="https://api.github.com/graphql",
        headers={
            "Authorization": f"Bearer {access_token}",
            "Content-Type": "application/json",
            "Accept": "application/vnd.github.bane-preview+json",
        },
    )

    issue = await client.get_issue(id="I_kwDOIPDwls55yyZU")

    print(issue)


if __name__ == "__main__":
    asyncio.run(run())
```

## Environment

Python 3.11.6

**Comment by mat-sop** [comment]
Hey, I managed to reproduce both of these issues:
1. It Looks like we omit artificially added `__typename` in all fragment strings. 
2. In the latest release we removed all `model_rebuild` calls. I tested this change with fragments, but it seems that the issue only occurs for nested fragments. We need to restore some of the removed `model_rebuild` calls.

I will create a separate issue with a more accurate description for the 1. issue and let's keep the current one for 2.
